### PR TITLE
Stacking particles

### PR DESCRIPTION
**Pull request by takore05**
water particles now build on each other and on top of sand. can be improved in terms of lag but it works well enough

### CODE REVIEW

**Comment by joshdabosh** [comment]
Good work, but I kind of want to clean this code up before merging to master (just some refactoring), which I think I'll get to early tomorrow.

**Comment by joshdabosh** [comment]
see slack for demo vid

File: js/dropsystem.js [hunk 1]
```diff
@@ -54,4 +62,4 @@ class ParticleManager {
             pop();
         });
     }
-}
+}
```
Comment (CLiu13) [inline js/dropsystem.js:65]: All files should have a newline at the end :P

File: js/index.js [hunk 2]
```diff
@@ -48,16 +48,16 @@ function draw() {
...
     if (mouseIsPressed) {
-        if (mouseX > 0 && mouseX< selector.clientWidth && mouseY > 0 && mouseY < selector.clientHeight) {
+        if (mouseX > 0 && mouseX < selector.clientWidth && mouseY > 0 && mouseY < selector.clientHeight) {
             system.addNewParticle(mouseX, mouseY);
         }
     }
 
     system.simulate();
-}
+}
```
Comment (CLiu13) [inline js/index.js:63]: (same comment about newline at the end of the file)

File: js/sand.js [hunk 3]
```diff
@@ -3,29 +3,26 @@ class SandParticle {
     constructor(xval, yval) {
         this.x = xval;
         this.y = yval;
-        if (this.y < height) {
+        
+        /*if (this.y < height) {
```
Comment (CLiu13) [inline js/sand.js:7]: Do we still need this code? If not, might as well just delete it entirely - no point keeping it as a comment if it is unnecessary.

File: js/sand.js [hunk 4]
```diff
@@ -34,26 +31,28 @@ class SandParticle {
...
+            let below = get(this.x, this.y + 2);
+            let left = get(this.x - 2, this.y + 2);
+            let right = get(this.x + 2, this.y + 2);
 
             if (below[0] != 255 && left[0] != 255 && right[0] != 255) {
                 // allow for the edge cases to be handled by adding a strike system
                 // it only stops if the particle cannot move for 20 frames in a row.
                 this.stoppedStrikes++;
                 return;
+            } else if(below[0] != 255){
```
Comment (CLiu13) [inline js/sand.js:54]: Add a space between the ```)``` and ```{```, just for standard formatting rules.

File: js/water.js [hunk 5]
```diff
@@ -38,39 +33,59 @@ class WaterParticle {
...
+        // water color
         stroke(0, 220, 220);
+        // fill circle
         fill(0, 180, 225);
         circle(this.x, this.y, 2);
         pop();
     }
-}
+
+}
```
Comment (CLiu13) [inline js/water.js:91]: (same comment about newline at the end of the file)

File: js/sand.js [hunk 6]
```diff
@@ -78,6 +77,4 @@ class SandParticle {
         circle(this.x, this.y, 2);
         pop();
     }
-}
-
-this.partiallyStopped = false;
+}
```
Comment (CLiu13) [inline js/sand.js:80]: (same comment about newline at the end of the file)

File: js/water.js [hunk 7]
```diff
@@ -5,31 +5,26 @@ class WaterParticle {
         // also creates speed property of how fast the droplets move back and forth 
         this.x = xval;
         this.y = yval;
+        /*
```
Comment (CLiu13) [inline js/water.js:8]: (same comment about removing unnecessary code)

**Comment by CLiu13** [comment]
Going to merge now since 2 team members have looked over this already (so that technically counts as an additional review).

**Pull request merged**